### PR TITLE
Add routerify integration

### LINKED ISSUE #484: [feature] Routerify
Nice lib, would be awesome to get askama support for routerify

### PR DESCRIPTION
**Pull request by dvtkrlbs**
closes #484

### CODE REVIEW

**Comment by dvtkrlbs** [comment]
This can also be generalized as a hyper integration since there is not any routerify specific code here.

File: askama_routerify/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+authors = ["Dirkjan Ochtman <dirkjan@ochtman.nl>"]
+description = "Routerify integration for Askama templates"
+homepage = "https://github.com/djc/askama"
+repository = "https://github.com/djc/askama"
+license = "MIT/Apache-2.0"
+workspace = ".."
+readme = "README.md"
+edition = "2018"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (djc) [inline askama_routerify/Cargo.toml:13]: Please get rid of this.

File: askama_shared/src/generator.rs [hunk 2]
```diff
@@ -361,6 +364,15 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
...
+    fn impl_routerify_impl(&mut self, buf: &mut Buffer) -> Result<(), CompileError> {
+        self.write_header(buf, "Into<::hyper::Response<::hyper::Body>>", None)?;
+        buf.writeln("fn into(self) -> hyper::Response<hyper::Body> {")?;
+
+        let ext = self.input.extension().unwrap_or("txt");
+        buf.writeln(&format!("::askama_routerify::respond(&self, {:?})", ext))?;
+        buf.writeln("}")?;
+        buf.writeln("}")
+    }
     // Writes header for the `impl` for `TraitFromPathName` or `Template`
```
Comment (djc) [inline askama_shared/src/generator.rs:376]: Add an empty line here below the new method?

**Review by djc — COMMENTED** [review]
Since the implementation does not require routerify at all, let's call the crate `askama_hyper` and rename all the bits in other Askama crates to `hyper`. Then rename the test file to `routerify.rs` (and please just merge `support.rs` into it.)

File: askama_routerify/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+[dependencies]
+askama = { version = "0.10.6", path = "../askama", default-features = false, features = ["with-routerify", "mime", "mime_guess"] }
+hyper = "0.14"
+routerify = "2.1"
```
[the remark this answers is not in the document]
Comment (dvtkrlbs) [reply askama_routerify/Cargo.toml:18]: If we turn this to askama_hyper I think we can drop it completely

[the remark this answers is not in the document]
Comment (djc) [reply askama_routerify/Cargo.toml:18]: Works for me!

**Comment by dvtkrlbs** [comment]
Did most of the stuff but the problem is we need the client feature of hyper on testing. It looks like testing only features is a open issue on cargo repo. So I don't know how to test the hyper code. Any idea?

**Comment by djc** [comment]
Sorry, not sure what you mean, can you give me more context for what you're trying to do?

**Comment by dvtkrlbs** [comment]
We test the http framework integrations and testing actix one is kinda hard since we need some kind of client or optionally activate the client feature on tests (i dont think this one is possible).

**Comment by djc** [comment]
You can just repeat the dependency in the dev-dependencies with the extra feature added?

**Pull request closed without merging**